### PR TITLE
verify-flags-underscore.py: Even fewer dash false positives

### PR DESCRIPTION
**Pull request by eparis**
- Fix a couple of docs (not actually false positives!)
- Better job ignore bash variables
- Better job ignoring jinja2 variables
- Better job ignoring salt variables

### CODE REVIEW

**Comment by eparis** [comment]
All total this is an 85% decrease in the size of the whitelist. While I can probably find a couple of more patterns to exclude the likelyhood of more false positives is quickly diminishing. The best thing to do at this point would be to convert the e2e tool to use pflag and flag normalization....

**Comment by eparis** [comment]
@krousey you are tired of me by now, right?  I promise, I'm done.

**Comment by krousey** [comment]
> @krousey you are tired of me by now, right? I promise, I'm done.

:tired_face:


...


Kidding.

**Comment by eparis** [comment]
Rebased and ready.  Should be easy to review this time.

File: docs/getting-started-guides/mesos.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ Update your PATH to more easily run the Kubernetes-Mesos binaries:
 export PATH="$(pwd)/_output/local/go/bin:$PATH"
 ```
 
-Identify your Mesos master: depending on your Mesos installation this is either a `host:port` like `mesos_master:5050` or a ZooKeeper URL like `zk://zookeeper:2181/mesos`.
+Identify your Mesos master: depending on your Mesos installation this is either a `host:port` like `mesos-master:5050` or a ZooKeeper URL like `zk://zookeeper:2181/mesos`.
```
Comment (krousey) [inline docs/getting-started-guides/mesos.md]: Are we sure about the changes in this file?

File: docs/getting-started-guides/mesos.md [hunk 2]
```diff
@@ -208,7 +208,7 @@ kubernetes       component=apiserver,provider=kubernetes   <none>     10.10.10.1
 ```
 
 Lastly, look for Kubernetes in the Mesos web GUI by pointing your browser to
-`http://<mesos_master_ip:port>`. Make sure you have an active VPN connection.
+`http://<mesos-master_ip:port>`. Make sure you have an active VPN connection.
```
Comment (krousey) [inline docs/getting-started-guides/mesos.md]: This seems highly suspect.

Comment (eparis) [inline docs/getting-started-guides/mesos.md]: this is...

File: docs/getting-started-guides/mesos.md (hunk 1 above)
Comment (eparis) [inline docs/getting-started-guides/mesos.md]: `mesos-master` `auth-path` `km-path` `executor-path` `--proxy-path` I verified are defined by mesos with a `-` and the docs are in fact wrong.

In this case mesos-master is purely a user indicator, and in fact they use mesos-master in other places.

File: pkg/util/logs.go [hunk 3]
```diff
@@ -25,8 +25,7 @@ import (
 	"github.com/spf13/pflag"
 )
 
-// Uses _ instead of - to better align with glog flags.
-var logFlushFreq = pflag.Duration("log_flush_frequency", 5*time.Second, "Maximum number of seconds between log flushes")
+var logFlushFreq = pflag.Duration("log-flush-frequency", 5*time.Second, "Maximum number of seconds between log flushes")
```
Comment (krousey) [inline pkg/util/logs.go]: @bgrant0607 Is this flag change ok?

File: hack/verify-flags-underscore.py [hunk 4]
```diff
@@ -95,8 +95,23 @@ def normalize_files(rootdir, files):
...
+        if "pillar[" + result + "]" in line:
+            return False
+        if "grains" + result in line:
+            return False
+        # These are usually yaml definitions
+        if result.endswith(":"):
+            return False
+         # something common in juju variables...
+        if "template_data[" + result + "]" in line:
+            return False
```
Comment (krousey) [inline hack/verify-flags-underscore.py]: Is it likely that this will continue to expand?

File: pkg/util/logs.go (hunk 3 above)
Comment (bgrant0607) [inline pkg/util/logs.go]: If it will still accept underscores, yes.

Comment (krousey) [inline pkg/util/logs.go]: @eparis will it?

Comment (eparis) [inline pkg/util/logs.go]: It will for all of our tools, yes.

File: hack/verify-flags-underscore.py (hunk 4 above)
Comment (eparis) [inline hack/verify-flags-underscore.py]: Not by me...   I ran out of things that could cut down the whitelist significantly.  But anyone who looks at the exceptions list might find more ideas....

**Pull request merged**